scripts/linters/pylint_extensions_test.py

```diff
@@ -3272,3 +3272,103 @@ def some_random_function(param1, param2):
+            msg_id='string-concatenation', args=('\'string1\''),
+            node=binop_node2)
+        left_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string2\''),
+            node=binop_node3)
+
+        with self.checker_test_object.assertAddsMessages(
+            str_str_message, right_message, left_message):
+            self.checker_test_object.checker.visit_binop(binop_node1)
+            self.checker_test_object.checker.visit_binop(binop_node2)
+            self.checker_test_object.checker.visit_binop(binop_node3)
+
+    def test_string_interpolation_does_not_raise_error(self):
+        binop_node1, binop_node2 = astroid.extract_node(
+            """
+            'string1%s' % (string2) #@
+            '%sstring2' % d #@
+            """)
+
+        with self.checker_test_object.assertNoMessages():
+            self.checker_test_object.checker.visit_binop(binop_node1)
+            self.checker_test_object.checker.visit_binop(binop_node2)
+
+    def test_addition_of_non_string_variables_does_not_raise_error(self):
+        binop_node1 = astroid.extract_node(
+            """
+            a = 2
+            b = 4
+            a + b #@
+            """)
+        with self.checker_test_object.assertNoMessages():
+            self.checker_test_object.checker.visit_binop(binop_node1)
+
+    def test_addition_of_string_variables_does_not_raise_error(self):
+        binop_node1 = astroid.extract_node(
+            """
+            a = 'string1'
+            b = 'string2'
+            a + b #@
+            """)
+        with self.checker_test_object.assertNoMessages():
+            self.checker_test_object.checker.visit_binop(binop_node1)
+
+    def test_multiple_string_concatenations_raises_error(self):
+        binop_node1 = astroid.extract_node(
+            """
+            'string1' + 'string2' + 'string3'
+            """)
+
+        binop_node2 = binop_node1.left
+        first_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string3\''),
+            node=binop_node1)
+        sec_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string1\''),
+            node=binop_node2)
+
+        with self.checker_test_object.assertAddsMessages(
+            first_message, sec_message):
+            self.checker_test_object.checker.visit_binop(binop_node1)
+            self.checker_test_object.checker.visit_binop(binop_node2)
+
+    def test_string_concatenation_inside_interpolation_raises_error(self):
+        mod_binop_node = astroid.extract_node(
+            """
+            a = 'string2'
+            'string1%s' % (a + 'string3') #@
+            """)
+
+        plus_binop_node = mod_binop_node.right
+        first_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string3\''),
+            node=plus_binop_node)
+
+        with self.checker_test_object.assertAddsMessages(first_message):
```

Comment:
It's not clear - are you raising an error for interpolation (i.e mod `%`) too? (We don't want to!)

Please follow the suggestion in previous comment to split the assert statements to make this clear.

Reply:
Yes understood, Implemented.